### PR TITLE
Fix broken jnp.nancumsum() & jnp.nancumprod() and add tests

### PR DESCRIPTION
**Pull request by jakevdp**

### CODE REVIEW

**Review by hawkinsp — COMMENTED** [review]
The change looks fine but you have test failures.

**Comment by jakevdp** [comment]
failures fixed

File: tests/lax_numpy_test.py [hunk 1]
```diff
@@ -1368,6 +1368,35 @@ def testCumSumProd(self, axis, shape, dtype, out_dtype, np_op, jnp_op, rng_facto
...
+    jnp_fun = partial(jnp_op, axis=axis, dtype=out_dtype)
+    jnp_fun = jtu.ignore_warning(category=jnp.ComplexWarning)(jnp_fun)
+
+    args_maker = lambda: [rng(shape, dtype)]
+
+    tol_thresholds = {dtypes.bfloat16: 4e-2}
+    tol = max(jtu.tolerance(dtype, tol_thresholds),
+              jtu.tolerance(out_dtype, tol_thresholds))
+    if dtype != jnp.bfloat16:
+      # numpy functions do not properly handle bfloat16
```
Comment (hawkinsp) [inline tests/lax_numpy_test.py:1395]: Some of them do work, but there's only so far we were able to go via a NumPy extension and without modifying NumPy itself.

**Review by hawkinsp — APPROVED** [review]

File: tests/lax_numpy_test.py (hunk 1 above)
Comment (jakevdp) [reply tests/lax_numpy_test.py:1395]: I was referring specifically to the functions being tested here, ``nancumsum`` and ``nancumprod``.

**Pull request merged**